tools/templates/moar/Makefile.in

```diff
@@ -270,22 +295,45 @@ $(NQP_MOAR): $(M_STAGE2_OUTPUT)
 	$(CP) @nfpq($(M_STAGE2_DIR)/$(P6QREGEX_MOAR))@ .
 	$(CP) @nfpq($(M_STAGE2_DIR)/$(NQP_MOAR))@ .
 
-$(M_RUNNER): @@configure_script@@ @@template(@platform@/nqp-m-build)@@
-	$(PERL5) @shquot(@configure_script@)@ @configure_opts()@ --expand=@template(@platform@/nqp-m-build)@ --out=@q($(M_RUNNER))@
-	$(CHMOD) 0755 @q($(M_RUNNER))@
-
-m-runner-default: m-all
-	$(CP) @q($(M_RUNNER))@ @nfpq($(BASE_DIR)/nqp$(BAT))@
-	$(CHMOD) 0755 nqp$(BAT)
+$(M_INST_NQP): @nfp(src/vm/moar/runner/main.c)@
+	$(RM_F) $@
+# Using only the pkgconfig moar includes does not work, because moar.h assumes all the specific includes below.
+	@moar::cc@ @moar::ccswitch@ @if(relocatable -DSTATIC_EXEC_PATH=@q(@c_escape(@nfp(@prefix@/bin/$(M_NQP))@)@)@)@ @static_nqp_home_define@ @moar::cflags@ $(M_MOAR_INC_PATHS) @moar::ccout@main@moar::obj@ @nfp(src/vm/moar/runner/main.c)@
```

Comment:
For maintainability purpose I would extract the command line options part which is being re-used below and make it one more variable. Same applies to `ld`.

Reply:
Done.